src/renderer/store/SsoAuthProvider.js

```diff
@@ -3,112 +3,16 @@
 import * as strings from '../../strings';
-import { logger } from '../../util/logger';
-import { extractJwtPayload } from '../auth/authUtil';
-import { AuthClient } from '../auth/clients/AuthClient';
-
-const { Util } = Common;
-
-//
-// Store
-//
 
 class SsoAuthProviderStore extends ProviderStore {
   // basic store is all that is needed for now
 }
 
 const pr = new SsoAuthProviderStore();
 
-//
-// Internal Methods
-//
-
-/**
- * [ASYNC] Start authorization with MCC to get the temp access code via the
- *  redirect URI that will use the 'lens://' protocol to redirect the user
- *  to Lens and ultimately call back into `_finishAuthorization()`.
- * @param {Object} options
- * @param {Object} options.config MCC Config object.
- */
-const _startAuthorization = async function ({ config }) {
-  pr.reset(true);
-
-  const authClient = new AuthClient({ config });
-
-  if (config.keycloakLogin) {
-    const url = authClient.getSsoAuthUrl();
-    Util.openExternal(url); // open in default browser
-  } else {
-    pr.loading = false;
-    pr.loaded = true;
-    pr.error = strings.ssoAuthProvider.error.basicOnly();
-  }
-
-  pr.notifyIfError();
-  pr.onChange();
-};
-
-/**
- * [ASYNC] Completes the authorization process by exchanging the temp access code
- *  for access tokens.
- *
- * NOTE: This method ASSUMES `loading` is `true` (i.e. load is in progress).
- *
- * @param {Object} options
- * @param {Object} options.oAuth OAuth response data from request for auth code.
- *  See `extEventOauthCodeTs` typeset in `eventBus.ts` for expected shape.
- * @param {Object} options.config MCC Config object.
- * @param {Cloud} options.cloud Current authentication information.
- *  This instance WILL be cleared and updated with new tokens.
- */
-const _finishAuthorization = async function ({ oAuth, config, cloud }) {
-  if (!pr.loading) {
-    // ignore rogue request to complete auth if it was canceled in Lens, but then
-    //  user completed request in browser for some reason
-    return;
-  }
-
-  const authClient = new AuthClient({ config });
-  let body;
-  let error;
-
-  if (oAuth.code) {
-    ({ body, error } = await authClient.getToken({ authCode: oAuth.code }));
-  } else {
-    // no code, something went wrong
-    error = oAuth.error || oAuth.error_description || 'unknown';
-  }
-
-  pr.loading = false;
-  pr.loaded = true;
-
-  if (error) {
-    logger.error(
-      'SsoAuthProvider._finishAuthorization()',
-      `Failed to get tokens from authorization code, error="${error}"`
-    );
-    pr.error = strings.ssoAuthProvider.error.authCode();
-  } else {
-    const jwt = extractJwtPayload(body.id_token);
-    if (jwt.preferred_username) {
-      cloud.updateTokens(body);
-      cloud.username = jwt.preferred_username;
-    } else {
-      logger.error(
-        'SsoAuthProvider._finishAuthorization()',
-        'Failed to get username from token JWT'
-      );
-      pr.error = strings.ssoAuthProvider.error.authCode();
-    }
-  }
```

Comment:
So now here, everything here that is _not_ setting state goes away because the utility method is handling it. But state still needs to be handled here, otherwise everything breaks.

```suggestion
  try {
    ssoUtil.finishAuthorization({ oAuth, config, cloud });
  } catch {
    pr.error = strings.ssoAuthProvider.error.authCode();
  }
```

Reply:
We can get the error fro catch like that:
`try {
            finishAuthorization(options);
          } catch (error) {
            pr.error = error;
          }`

Reply:
Yes, but the point is that we're generating a __user-facing__ error message, not just giving them the "cyptic" generated error message. So we want to use our own (more generic, but still specific enough) error string instead of the "raw" error message.

Also note that the error object here is an __object__, not a string, and `pr.error` is a string. But regardless, we want to use our own user-facing string.